src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_account_scenarios.py

```diff
@@ -643,3 +643,43 @@ def test_storage_account_update_change_feed(self):
 
         result = self.cmd('storage account blob-service-properties show -n {sa} -g {rg}').get_output_in_json()
         self.assertEqual(result['changeFeed']['enabled'], True)
+
+    @ResourceGroupPreparer(name_prefix='cli_storage_account_update_delete_retention_policy')
+    @StorageAccountPreparer()
+    def test_storage_account_update_delete_retention_policy(self, resource_group, storage_account):
+        self.kwargs.update({
+            'sa': storage_account,
+            'rg': resource_group
+        })
+
+        result = self.cmd('storage account blob-service-properties update --delete-retention false -n {sa} -g {rg}').get_output_in_json()
+        self.assertEqual(result['deleteRetentionPolicy']['enabled'], False)
+
+        with self.assertRaises(SystemExit):
+            self.cmd('storage account blob-service-properties update --delete-retention true -n {sa} -g {rg}')
+
+        result = self.cmd('storage account blob-service-properties update --delete-retention-period 1 -n {sa} -g {rg}').get_output_in_json()
+        self.assertEqual(result['deleteRetentionPolicy']['days'], 1)
+
+        result = self.cmd('storage account blob-service-properties update --delete-retention-period 365 -n {sa} -g {rg}').get_output_in_json()
+        self.assertEqual(result['deleteRetentionPolicy']['days'], 365)
+
+        result = self.cmd('storage account blob-service-properties update --delete-retention-period 100 -n {sa} -g {rg}').get_output_in_json()
+        self.assertEqual(result['deleteRetentionPolicy']['days'], 100)
+
+        with self.assertRaises(SystemExit):
+            self.cmd('storage account blob-service-properties update --delete-retention-period -1 -n {sa} -g {rg}')
+
+        with self.assertRaises(SystemExit):
+            self.cmd('storage account blob-service-properties update --delete-retention-period 0 -n {sa} -g {rg}')
+
+        with self.assertRaises(SystemExit):
+            self.cmd('storage account blob-service-properties update --delete-retention-period 366 -n {sa} -g {rg}')
+
+        result = self.cmd('storage account blob-service-properties update --delete-retention false --delete-retention-period 365 -n {sa} -g {rg}').get_output_in_json()
```

Comment:
When disabling delete retention policy, users don't need to provide --delete-retention-period parameter, right?

Reply:
Yes, I have changed it